mlflow/gateway/providers/bedrock.py

```diff
@@ -0,0 +1,279 @@
+import json
+import os
+import time
+from enum import Enum
+
+import boto3
+import botocore.config
+import botocore.exceptions
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.config import AWSBedrockConfig, AWSIdAndKey, AWSRole, RouteConfig
+from mlflow.gateway.constants import (
+    MLFLOW_AI_GATEWAY_ANTHROPIC_DEFAULT_MAX_TOKENS,
+    MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS,
+)
+from mlflow.gateway.exceptions import AIGatewayConfigException
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import rename_payload_keys, send_request
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class AWSBedrockModelProvider(Enum):
+    AMAZON = "amazon"
+    COHERE = "cohere"
+    AI21 = "ai21"
+    ANTHROPIC = "anthropic"
+
+    @classmethod
+    def of_str(cls, name: str):
+        name = name.lower()
+
+        for opt in cls:
+            if opt.name.lower() == name or opt.value.lower() == name:
+                return opt
+
+
+class AWSBedrockProvider(BaseProvider):
+    def __init__(self, config: RouteConfig):
+        super().__init__(config)
+
+        if config.model.config is None or not isinstance(config.model.config, AWSBedrockConfig):
+            raise TypeError(f"Invalid config type {config.model.config}")
+        self.bedrock_config: AWSBedrockConfig = config.model.config
+        self._client = None
+        self._client_created = 0
+
+    def _client_expired(self):
+        if not isinstance(self.bedrock_config.aws_config, AWSRole):
+            return False
+
+        return (
+            (time.monotonic_ns() - self._client_created)
+            >= (self.bedrock_config.aws_config.session_length_seconds) * 1_000_000_000,
+        )
+
+    def get_bedrock_client(self):
+        if self._client is not None and not self._client_expired():
+            return self._client
+
+        session_args, config_args, client_args = {}, {}, {}
+        aws_config = self.bedrock_config.aws_config
+
+        region_name = aws_config.aws_region or os.environ.get(
+            "AWS_REGION", os.environ.get("AWS_DEFAULT_REGION")
+        )
+
+        if region_name:
+            config_args["region_name"] = session_args["region_name"] = region_name
+
+        # check profile, mostly for dev
+        profile_name = os.environ.get("AWS_PROFILE")
+        if profile_name:
+            session_args["profile_name"] = profile_name
```

Comment:
Seems like this will override what's currently in the boto config. Should we add "profile" to the config YAML instead?

Reply:
@dbczumar I'm not sure I follow.  You're thinking this could/should live in the route definition, and be picked up by AWSBedrockConfig like the key/role variables?  Or is there another point at which Gateway is setting / picking up an AWS profile that I'm unknowingly clobbering here?